qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -0,0 +1,287 @@
+from .exceptions import IBMQJobManagerInvalidStateError, IBMQJobManagerTimeoutError
+from ..job import IBMQJob
+from ..ibmqbackend import IBMQBackend
+
+logger = logging.getLogger(__name__)
+
+
+class ManagedJobSet:
+    """A set of managed jobs."""
+
+    def __init__(self, name: Optional[str] = None) -> None:
+        """Creates a new ManagedJobSet instance."""
+        self._managed_jobs = []  # type: List[ManagedJob]
+        self._name = name or datetime.utcnow().isoformat()
+
+        # Used for caching
+        self._results = []  # type: Optional[List[Union[Result, None]]]
+        self._error_msg = None  # type: Optional[str]
+
+    def run(
+            self,
+            experiment_list: Union[List[List[QuantumCircuit]], List[List[Schedule]]],
+            backend: IBMQBackend,
+            executor: ThreadPoolExecutor,
+            **assemble_config: Any
+    ) -> None:
+        """Execute a list of circuits or pulse schedules on a backend.
+
+        Args:
+            experiment_list : Circuit(s) or pulse schedule(s) to execute.
+            backend: Backend to execute the experiments on.
+            executor: The thread pool to use.
+            assemble_config: Additional arguments used to configure the Qobj
+                assembly. Refer to the ``qiskit.compiler.assemble`` documentation
+                for details on these arguments.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If the jobs were already submitted.
+        """
+        if self._managed_jobs:
+            raise IBMQJobManagerInvalidStateError("Jobs were already submitted.")
+
+        exp_index = 0
+        for i, experiment in enumerate(experiment_list):
+            qobj = assemble(experiment, backend=backend, **assemble_config)
+            job_name = "{}_{}_".format(self._name, i)
+            future = executor.submit(
+                self._async_submit, qobj=qobj, job_name=job_name, backend=backend)
+            self._managed_jobs.append(
+                ManagedJob(experiment, start_index=exp_index, future=future))
+            exp_index += len(experiment)
+
+    def _async_submit(
+            self,
+            qobj: Qobj,
+            job_name: str,
+            backend: IBMQBackend,
+    ) -> IBMQJob:
+        """Run a Qobj asynchronously.
+
+        Args:
+            qobj: Qobj to run.
+            job_name: Name of the job.
+            backend: Backend to execute the experiments on.
+
+        Returns:
+            IBMQJob instance for the job.
+        """
+        return backend.run(qobj=qobj, job_name=job_name)
+
+    def submit_results(self) -> None:
+        """Collect job submit responses.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If jobs have not been submitted.
+        """
+        if not self._managed_jobs:
+            raise IBMQJobManagerInvalidStateError("Jobs need to be submitted first!")
+
+        if all(mjob.future is None for mjob in self._managed_jobs):
```

Comment:
Can we find a different way of checking if the managed job has been submitted other than removing its `.future`?

Having the future around can be handy (for example, for some instances of cancelling) and I think we can assume the future is always in progress (either waiting for a spot or running) - maybe `future.done()` can help. If no good alternatives are found, we can have some sort of flag at the `ManagedJobSet` level (ie. `ManagedJobSet()._i_have_really_collected_everything = True`.